src/store/reducers/tableData.ts

```diff
@@ -0,0 +1,38 @@
+import type {BaseQueryFn, EndpointBuilder} from '@reduxjs/toolkit/query';
+
+import type {FetchData, PaginatedTableData, SortParams} from '../../components/PaginatedTable';
+
+import {api} from './api';
+
+interface PaginatedTableParams<T, F> {
+    id: number;
+    fetchData: FetchData<T, F>;
+    filters: F;
+    chunkSize: number;
+    sortParams?: SortParams;
+}
+
+function endpoints<T, F>(build: EndpointBuilder<BaseQueryFn, string, string>) {
+    return {
+        fetchTableChunk: build.query<PaginatedTableData<T>, PaginatedTableParams<T, F>>({
+            queryFn: async ({id, chunkSize, sortParams, filters, fetchData}, {signal}) => {
+                try {
+                    const offset = id * chunkSize;
+                    const response = await fetchData(
+                        chunkSize,
+                        offset,
+                        filters,
+                        sortParams,
+                        signal,
+                    );
+                    return {data: response};
+                } catch (error: unknown) {
+                    return {error};
+                }
+            },
+            providesTags: ['All'],
```

Comment:
Should we add some other params to `provideTags` for refetch tuning? Maybe some `key` related with `fetchData` to deal with this https://github.com/ydb-platform/ydb-embedded-ui/pull/1146/files#diff-3548c791e04083de8b490c582b83dd003f9b06c544dac512c85983a1fe16b86cR38 case?